src/ucp/core/ucp_ep.c

```diff
@@ -332,66 +373,8 @@ void ucp_ep_destroy_base(ucp_ep_h ep)
     ucs_strided_alloc_put(&ep->worker->ep_alloc, ep);
 }
 
-ucs_status_t ucp_worker_create_ep(ucp_worker_h worker, unsigned ep_init_flags,
-                                  const char *peer_name, const char *message,
-                                  ucp_ep_h *ep_p)
-{
-    ucp_context_h context = worker->context;
-    ucs_status_t status;
-    ucp_ep_h ep;
-
-    status = ucp_ep_create_base(worker, peer_name, message, &ep);
-    if (status != UCS_OK) {
-        goto err;
-    }
-
-    if (!(ep_init_flags & UCP_EP_INIT_FLAG_INTERNAL) &&
-        ((context->config.ext.proto_indirect_id == UCS_CONFIG_ON) ||
-         ((context->config.ext.proto_indirect_id == UCS_CONFIG_AUTO) &&
-          (ep_init_flags & UCP_EP_INIT_ERR_MODE_PEER_FAILURE)))) {
-        ucp_ep_update_flags(ep, UCP_EP_FLAG_INDIRECT_ID, 0);
-    }
-
-    status = UCS_PTR_MAP_PUT(ep, &worker->ep_map, ep,
-                             !!(ep->flags & UCP_EP_FLAG_INDIRECT_ID),
-                             &ucp_ep_ext_control(ep)->local_ep_id);
-    if ((status != UCS_OK) && (status != UCS_ERR_NO_PROGRESS)) {
-        ucs_error("ep %p: failed to allocate ID: %s", ep,
-                  ucs_status_string(status));
-        goto err_destroy_ep_base;
-    }
-
-    if (ep_init_flags & UCP_EP_INIT_FLAG_INTERNAL) {
-        ucp_ep_update_flags(ep, UCP_EP_FLAG_INTERNAL, 0);
-        ucs_list_add_tail(&worker->internal_eps, &ucp_ep_ext_gen(ep)->ep_list);
-    } else {
-        ucs_list_add_tail(&worker->all_eps, &ucp_ep_ext_gen(ep)->ep_list);
-        ucs_assert(ep->worker->num_all_eps < UINT_MAX);
-        ++ep->worker->num_all_eps;
-    }
-
-    ucp_ep_flush_state_reset(ep);
-
-    *ep_p = ep;
-    return UCS_OK;
-
-err_destroy_ep_base:
-    ucp_ep_refcount_assert(ep, create, ==, 1);
-    ucp_ep_refcount_remove(ep, create);
-err:
-    return status;
-}
-
 void ucp_ep_delete(ucp_ep_h ep)
```

Comment:
this func is called in error flows of many other functions. Shall it be replaced with `ucp_ep_destroy_base` now? Do we need to keep this func at all?